assignment1/string_permutation.py

```diff
@@ -0,0 +1,18 @@
+import string 
+
+def string_permutation(str1, str2):
+    if (str.isalpha(str1) or str1 == "") and (
```

Comment:
Purely a style comment, but to make function bodies easier to read, it is generally better to do all the checks up front and then proceed with the main part of the function body as top-level code, that is:

```
def string_permutation(str1, str2):
  if not ((str.isalpha(str1) or str1 == "") and (str.isalpha(str2) or str2 == "")):
    raise ValueError(...)
  str1 = str1.lower()
  ...
```

It does not always matter, but tends to replace in easier to read ans "shallower" code. Especially if multiple layers of conditionals and loops are involved.